File: fsharp/isogram/Isogram.fs
module Isogram

    open System

    let isIsogram (str:string):bool =
        let characters:char array =
            str.ToLower().ToCharArray()
            |> Array.filter (Char.IsLetter)
        let length:int =
            characters
            |> Array.length
        let distinctLength:int =
            characters
            |> Array.distinct
            |> Array.length
        length = distinctLength

File: solutions/scala/isogram/1/src/main/scala/Isogram.scala
object Isogram
{
    def isIsogram (str:String):Boolean =
    {
        val characters:Array[Char] = str.toLowerCase().toCharArray().filter(Character.isLetter)
        val length:Int = characters.length
        val distinctLength:Int = characters.distinct.length
        length == distinctLength
    }
}